arrays/arrays.c

```diff
@@ -101,10 +130,16 @@ void arr_append(Array *arr, char *element) {
 
   // Resize the array if the number of elements is over capacity
   // or throw an error if resize isn't implemented yet.
+  if (arr->count == arr->capacity) {
+    resize_array(arr);
+  }
 
   // Copy the element and add it to the end of the array
+  char *new_element = element;
```

Comment:
Should be char *new_element = strdup(element); That's in order to malloc memory for the string. Otherwise there's no point to doing new_element = element. You might as well put element directly into arr->elements[arr->count]